server/data/data-memory/src/main/java/org/apache/james/user/memory/MemoryUsersRepository.java

```diff
@@ -115,9 +115,9 @@ public int countUsers() throws UsersRepositoryException {
 
     @Override
     public Iterator<Username> list() throws UsersRepositoryException {
-        return userByName.values()
+        return userByName.keySet()
             .stream()
-            .map(User::getUserName)
+            .map(Username::of)
```

Comment:
Username type can be used as a key instead of string?

Reply:
could be, but then it would be time to enforce Username to always be lower case, which is not assumed in our code base. For example in Cassandra implementation we store a lower case Username as Id, and the case sensitive version in Name.

Btw I made this change for correctly supporting no virtual hosting.

Reply:
oops it seems to not fix filterShouldAddUsernameWhenNoVirtualHostingAndMailboxSessionContainsUsername finally... Maybe the simpler would be to remove this test...

Reply:
Agreed. Validity of username in regard of usersRepository should be done before, with mailboxSession opening

Reply:
See https://github.com/linagora/james-project/pull/2815/commits/742b0e8b4ba36b80914a870b2b88fc49ccc08c07